devtools/src/index.ts

```diff
@@ -25,3 +26,26 @@ export function provideQueryDevTools(
     },
   ]);
 }
+
+export function provideLazyQueryDevTools(
+  devToolOptions: Partial<Omit<TanstackQueryDevtoolsConfig, 'client'>> = {},
+  triggerFunction: string = 'toggleDevtools',
+) {
+  return makeEnvironmentProviders([
+    {
+      provide: APP_INITIALIZER,
```

Comment:
Any reason why you chose `APP_INITIALIZER` instead of `ENVIRONMENT_INITIALIZER`? 
Using latter one lets you provide the dev-tools in a child route, using the `APP_INITIALIZER` only lets you inject it into the start of the whole application, wouldn't it be better to just merge it into the already provided `provideQueryDevTools`? 
But nonetheless i dont see it as a good fit bringing devtools into production. TanStack isn't providing a function for that either, at the end it is up to the user to provide them in there.

Reply:
Injecting the query client wasn't working in `ENVIRONMENT_INITIALIZER` - unless I had the `inject()` call in the wrong place somehow.